swarm/pss/pss_test.go

```diff
@@ -482,29 +486,32 @@ func testAsymSend(t *testing.T) {
 	}
 }
 
-// params in run name:
-// nodes/msgs/addrbytes/adaptertype
-// if adaptertype is exec uses execadapter, simadapter otherwise
-func TestNetwork(t *testing.T) {
-	//t.Run("8/1024/2/exec", testNetwork)
-	t.Run("256/128/4/sock", testNetwork)
+type Job struct {
+	Msg      []byte
+	SendNode discover.NodeID
+	RecvNode discover.NodeID
 }
 
-func testNetwork(t *testing.T) {
+func worker(id int, jobs <-chan Job, rpcs map[discover.NodeID]*rpc.Client, pubkeys map[discover.NodeID][]byte, hextopic string) {
+	for j := range jobs {
+		rpcs[j.SendNode].Call(nil, "pss_sendAsym", common.ToHex(pubkeys[j.RecvNode]), hextopic, j.Msg)
+	}
+}
 
+func TestNetwork(t *testing.T) {
 	type msgnotifyC struct {
 		id     discover.NodeID
 		msgIdx int
 	}
 	topic := whisper.BytesToTopic([]byte("foo:42"))
 	hextopic := common.ToHex(topic[:])
 
-	paramstring := strings.Split(t.Name(), "/")
-	nodecount, _ := strconv.ParseInt(paramstring[1], 10, 0)
-	msgcount, _ := strconv.ParseInt(paramstring[2], 10, 0)
-	addrsize, _ := strconv.ParseInt(paramstring[3], 10, 0)
-	messagedelayvarianceusec := (int(msgcount)/1000 + 1) * 1000 * 1000
-	log.Info("network test", "nodecount", nodecount, "msgcount", msgcount, "addrhintsize", addrsize, "sendtimevariance", messagedelayvarianceusec/(1000*1000))
+	nodecount := *fnodecount
+	msgcount := *fmsgs
+	addrsize := *faddrsize
+	adapter := *fadapter
+
+	log.Info("network test", "nodecount", nodecount, "msgcount", msgcount, "addrhintsize", addrsize)
 
```

Comment:
You've removed the message delay. Is the intention to have all messages will race to send at the same time?

Reply:
Yes, I removed the delays, because the different adapters behaved differently and I wanted to increase concurrency. With delays tests were failing at different thresholds for me.

Reply:
That is interesting in itself, though? Why were they failing with delays, you think?

Reply:
They were failing with or without delays. It was more difficult to reproduce with the delays though, because it introduces more non-determinism.

At some point when I increased the delays a lot, the tests stopped failing - essentially because I removed the concurrency.